Snippets/Snippets_6/Callback/Usage.cs

```diff
@@ -0,0 +1,32 @@
+﻿namespace Snippets6.Callback
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+
+    public class Usage
+    {
+        public async Task Describe()
+        {
+            IBus bus = null;
+            var options = new SendOptions();
+
+            #region CallbackWithMessageAsResponse
+            var responseMessage = await bus.RequestWithTransientlyHandledResponseAsync<PlaceOrderResponse>(new PlaceOrder(), options);
```

Comment:
Started an internal poll. I think we should comply to the "rules"

Quote from Steven Toub, PFX team

> If a public method is Task-returning and is asynchronous in nature (as opposed to a method that is known to always execute synchronously to completion but still returns a Task for some reason), it should have an “Async” suffix. That’s the guideline. The primary goal here with the naming is to make it very obvious to a consumer of the functionality that the method being invoked will likely not complete all of its work synchronously; it of course also helps with the case where functionality is exposed with both synchronous and asynchronous methods such that you need a name difference to distinguish them. How the method achieves its asynchronous implementation is immaterial to the naming: whether async/await is used to garner the compiler’s help, or whether types and methods from System.Threading.Tasks are used directly (e.g. TaskCompletionSource) doesn’t really matter, as that doesn’t affect the method’s signature as far as a consumer of the method is concerned.

> Of course, there are always exceptions to a guideline. The most notable one in the case of naming would be cases where an entire type’s raison d’etre is to provide async-focused functionality, in which case having Async on every method would be overkill, e.g. the methods on Task itself that produce other Tasks.

> As for void-returning asynchronous methods, it’s not desirable to have those in public surface area, since the caller has no good way of knowing when the asynchronous work has completed. If you must expose a void-returning asynchronous method publicly, though, you likely do want to have a name that conveys that asynchronous work is being initiated, and you could use the “Async” suffix here if it made sense. Given how rare this case should be, I’d argue it’s really a case-by-case kind of decision.

> I hope that helps, Steve

[Official naming convention:](https://msdn.microsoft.com/en-us/library/hh191443.aspx#BKMK_NamingConvention)